builtin/aws/ecs/task.go

```diff
@@ -0,0 +1,245 @@
+func (p *TaskLauncher) StopTask(
+	ctx context.Context,
+	log hclog.Logger,
+	ti *TaskInfo,
+) error {
+	return nil
+}
+
+// StartTask runs an ECS Task to perform the requested job.
+func (p *TaskLauncher) StartTask(
+	ctx context.Context,
+	log hclog.Logger,
+	tli *component.TaskLaunchInfo,
+) (*TaskInfo, error) {
+	// Generate an ID for our task name.
+	id, err := ulid.New(ulid.Now(), rand.Reader)
+	if err != nil {
+		return nil, err
+	}
+
+	// Arguments here represent the command inputs used when executing the
+	// waypoint runner, e.x:
+	//   "runner, agent, -vvv, -id, <some id>, -odr"
+	cmd := aws.StringSlice(tli.Arguments)
+
+	sess, err := utils.GetSession(&utils.SessionConfig{
+		Region: p.config.Region,
+	})
+	if err != nil {
+		return nil, err
+	}
+	ecsSvc := ecs.New(sess)
+
+	envs := []*ecs.KeyValuePair{}
+	for k, v := range tli.EnvironmentVariables {
+		envs = append(envs, &ecs.KeyValuePair{
+			Name:  aws.String(k),
+			Value: aws.String(v),
+		})
+	}
+	// This environment variable informs Kaniko that it is operating in a
+	// container. Without this environment variable, Kaniko will fail with an
+	// error
+	envs = append(envs, &ecs.KeyValuePair{
+		Name:  aws.String("container"),
+		Value: aws.String("docker"),
+	})
+
+	taskName := fmt.Sprintf("waypoint-odr-task-%s", id.String())
+
+	logOptions := buildLoggingOptions(
+		nil,
+		p.config.Region,
+		p.config.LogGroup,
+		taskName,
+	)
+
+	def := ecs.ContainerDefinition{
+		Name:        &taskName,
+		Image:       &tli.OciUrl,
+		Command:     cmd,
+		Environment: envs,
+		LogConfiguration: &ecs.LogConfiguration{
+			LogDriver: aws.String(ecs.LogDriverAwslogs),
+			Options:   logOptions,
+		},
+	}
+
+	exRoleArn, err := roleArn(p.config.ExecutionRoleName, sess)
+	if err != nil {
+		return nil, err
+	}
+
+	taskRoleArn, err := roleArn(p.config.OdrTaskRoleName, sess)
+	if err != nil {
+		return nil, err
+	}
+
+	registerTaskDefinitionInput := ecs.RegisterTaskDefinitionInput{
+		ContainerDefinitions:    []*ecs.ContainerDefinition{&def},
+		ExecutionRoleArn:        &exRoleArn,
+		TaskRoleArn:             &taskRoleArn,
+		Cpu:                     aws.String("1024"),
+		Memory:                  aws.String("2048"),
+		Family:                  aws.String("waypoint-runner"),
+		NetworkMode:             aws.String("awsvpc"),
+		RequiresCompatibilities: []*string{aws.String(ecs.LaunchTypeFargate)},
+	}
+
+	taskDef, err := utils.RegisterTaskDefinition(&registerTaskDefinitionInput, ecsSvc)
+	if err != nil {
+		return nil, err
+	}
+
+	taskDefArn := *taskDef.TaskDefinitionArn
+
+	subnetStrings := strings.Split(p.config.Subnets, ",")
+	subnets := aws.StringSlice(subnetStrings)
+
+	if _, err := ecsSvc.RunTask(&ecs.RunTaskInput{
+		LaunchType:           aws.String(ecs.LaunchTypeFargate),
+		Cluster:              &p.config.Cluster,
+		Count:                aws.Int64(1),
+		TaskDefinition:       &taskDefArn,
+		EnableECSManagedTags: aws.Bool(true),
```

Comment:
`EnableECSManagedTags` ensures any tasks / containers created by this service will inherit the tags of this service.